ranger-drove/src/main/java/io/appform/ranger/drove/servicefinderhub/DroveServiceDataSource.java

```diff
@@ -22,21 +22,26 @@
 import io.appform.ranger.drove.common.DroveNodeDataStoreConnector;
 import io.appform.ranger.drove.config.DroveUpstreamConfig;
 import io.appform.ranger.drove.common.DroveCommunicator;
+import java.util.Collections;
+import java.util.Set;
 import lombok.extern.slf4j.Slf4j;
 
 import java.util.Collection;
 
 @Slf4j
 public class DroveServiceDataSource<T> extends DroveNodeDataStoreConnector<T> implements ServiceDataSource {
     private final String namespace;
+    private final Set<String> excludedServices;
```

Comment:
This doesn't look correct. The filter should be done _after_ we get the list at the higher level. Let us not expect data source implementers to implement this.

Reply:
Revert to original interface and filter after the service list is returned from source in servicefinderhub, client etc. Those are in our control and removes the burden to filter from data source implementers